src/messages.rs

```diff
@@ -33,34 +34,44 @@ pub struct WebDriverError {
     pub stacktrace: Option<String>,
 }
 
+#[derive(Serialize, Default)]
+pub struct Capabilities {
+    alwaysMatch: BTreeMap<String, JsonValue>,
+}
+
 #[derive(Serialize)]
 pub struct NewSessionCmd {
-    capabilities: JsonValue,
+    capabilities: Capabilities,
 }
 
 impl NewSessionCmd {
     pub fn new() -> Self {
+        let mut capabilities: Capabilities = Default::default();
+        capabilities.alwaysMatch.insert("goog:chromeOptions".to_string(), json!({"w3c": true}));
         NewSessionCmd {
-            capabilities: json!({
-                "alwaysMatch": {
-                    // ask chrome to be w3c compliant
-                    "goog:chromeOptions": {
-                        "w3c": true
-                    }
-                }
-            }),
+            capabilities,
         }
     }
 
 // TODO firefox specifc prefs
 // [moz:firefoxOptions][prefs][name] = value;
 
+    pub fn capability(&mut self, name: &str, value: Option<JsonValue>) {
```

Comment:
Can we move this setter to `struct Capabilities` and rename it to `always`, `always_match` or similar? The [capabilities spec](https://w3c.github.io/webdriver/webdriver-spec.html#processing-capabilities) describes `alwaysMatch` and `firstMatch` here, which we may add later.

I really like passing the value as an Option<_> to allow insert or remove with one method. 👍

Reply:
Other possibilities here are

- a method to merge capabilities
- a build pattern of some kind, that either returns Self or &mut Self